src/utils.js

```diff
@@ -0,0 +1,204 @@
+class Task {
+  constructor(description, completed, index) {
+    this.description = description;
+    this.completed = completed;
+    this.index = index;
+  }
+}
+
+function restoreOriginalIcons() {
+  const iconDiv = document.querySelectorAll('.icon');
+  iconDiv.forEach((icon) => {
+    icon.innerHTML = '&#8942';
+    icon.style.cursor = 'all-scroll';
+  });
+}
+
+function restoreBackgroundWhite() {
+  const labelTasks = document.querySelectorAll('li');
+  labelTasks.forEach((label) => {
+    label.style['background-color'] = '#fff';
+  });
+}
+
+function scrapeAndSave() {
+  const tasksArray = [];
+  const tasks = document.querySelectorAll('.task');
+  tasks.forEach((label, index) => {
+    const parentTaskCont = label.parentNode;
+    const checkboxElem = parentTaskCont.querySelector('.checkBoxesTasks');
+    const task = new Task(label.innerHTML, checkboxElem.checked, index + 1);
+    tasksArray.push(task);
+  });
+  localStorage.removeItem('tasksList');
+  localStorage.setItem('tasksList', JSON.stringify(tasksArray));
+}
+
+function setValueTextArea(textArea, labelTask) {
+  labelTask.innerHTML = textArea.value;
+  textArea.style.display = 'none';
+  labelTask.style.display = 'block';
+  // restore original icons
+  restoreOriginalIcons();
+  // change background to white
+  restoreBackgroundWhite();
+  // Save to local storage
+  scrapeAndSave();
+}
+
+function setValueTextAreaAlls() {
+  const textAreas = document.querySelectorAll('.textArea');
+  textAreas.forEach((textArea) => {
+    const parentDivTaskCont = textArea.parentNode;
+    const labelTask = parentDivTaskCont.querySelector('.task');
+    if (textArea.style.display === 'block') {
+      setValueTextArea(textArea, labelTask);
+    }
+  });
+}
+
+export function addEventclearAllButton() {
+  const buttonClearAll = document.querySelector('.clearAll');
+  buttonClearAll.addEventListener('click', () => {
+    setValueTextAreaAlls();
+    const liElements = document.querySelectorAll('li');
+    liElements.forEach((liElem) => {
+      const checkboxElem = liElem.querySelector('.checkBoxesTasks');
+      if (checkboxElem.checked) {
+        liElem.remove();
+      }
```

Comment:
- [ ] kindly use the `Array.filter() method of javascript to implement `clear all button` feature as stated in the [requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md#project-requirements).

> - Implement a function for the "Clear all completed" button (use filter() method).